File: libexec/git-ediff-wrapper.pl
#!/usr/bin/env perl

use warnings;
use strict;
use FindBin qw($Bin);


$| = 1;

sub diff_emacs {
    my $diff_a = shift;
    my $diff_b = shift;

    my $zzemacs_path = "~/zzemacs";

    # Reset path on linux
    if ( "$^O" eq "linux" ) {
        $zzemacs_path = "$Bin/..";
    } elsif ( "$^O" eq "msys" ) {
        $zzemacs_path = "$ENV{ZZNIX_HOME}/home/zhoujd/zzemacs";
    }

    my $elisp_string = sprintf("(progn \
                                    (load-file \\\"%s/elisp/ediff-sample.el\\\") \
                                    (ediff-sample-diff \\\"%s\\\" \\\"%s\\\") \
                                )", $zzemacs_path, $diff_a, $diff_b);

    my $emacs_para = "-nw -Q";
    if ("$ENV{TERM}" eq "dumb") {
        $emacs_para = "-Q";
    }

    my $cmd = sprintf("emacs %s --eval \"%s\"", $emacs_para, $elisp_string);

    (system("$cmd") == 0) || die "Cannot run $cmd $!";
}

sub main {
    diff_emacs($ARGV[0], $ARGV[1]);
}

main;

File: libexec/git-ediff-wrapper.py
#!/usr/bin/env python

### parameter desc
#$child $parent

import os
import sys
import platform

def diff_emacs(a, b):
    strfilepath = os.path.realpath(__file__)
    sysstr = platform.system()
    if sysstr == "Windows":
        a = a.replace("\\", "/")
        b = b.replace("\\", "/")
        strfilepath = strfilepath.replace("\\", "/")

    zzemacs_path = "%s/../" % (os.path.dirname(strfilepath),)
    elisp_string="(progn \
                    (load-file \\\"%s/elisp/ediff-sample.el\\\") \
                    (ediff-sample-diff \\\"%s\\\" \\\"%s\\\") \
                  )" % (zzemacs_path, a, b)

    cmd = "emacs -q --no-site-file --eval \"%s\"" % elisp_string
    os.system(cmd)

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print "Using example: %s <diff-A> <diff-B>" % sys.argv[0]
        sys.exit(1)

    diff_emacs(sys.argv[1], sys.argv[2])
